Inedo.DependencyScan/NpmDependencyScanner.cs

```diff
@@ -22,13 +22,41 @@ public override async Task<IReadOnlyCollection<ScannedProject>> ResolveDependenc
 
         private static IEnumerable<DependencyPackage> ReadDependencies(JsonDocument doc)
         {
-            if (!doc.RootElement.TryGetProperty("dependencies", out var dependencies))
+            if (!doc.RootElement.TryGetProperty("lockfileVersion", out var lockFileVersion))
                 yield break;
 
-            foreach (var d in dependencies.EnumerateObject())
+            var lockFileVersionString = lockFileVersion.ToString();
```

Comment:
I considered getting the Number value / int but wanted to protect against any weirdly formatted values. Perhaps that is being to cautious.

Reply:
I wouldn't check the lockFileVersion at all, to be honest. Instead, I'd check for the property "packages" and fall back to reading the property "dependecies". Otherwise this code will break as soon as NPM releases a lockFileVersion 4.

Reply:
100% agree - I've updated to instead just check for the package and dependencies nodes themselves